src/Juno/Consensus/ByzRaft/Commit.hs

```diff
@@ -133,17 +132,16 @@ updateCommitIndex = do
                   commitProof %= Map.filter (\a -> qci < _aerIndex a)
                   debug $ "Commit index is now: " ++ show qci
                   return True
-                else do
-                  return False
+                else return False
 
-checkCommitProof :: Int -> Seq LogEntry -> Int -> [AppendEntriesResponse] -> Either Int LogIndex
+checkCommitProof :: Int -> Ledger LogEntry -> LogIndex -> [AppendEntriesResponse] -> Either Int LogIndex
 checkCommitProof qsize les maxLogIdx evidence = go 0 evidence
   where
     go n [] = Left n -- no update
-    go n (ev:evs) = if fromIntegral (_aerIndex ev) > maxLogIdx || fromIntegral (_aerIndex ev) < (0::Int)
+    go n (ev:evs) = if _aerIndex ev > maxLogIdx || _aerIndex ev < 0
```

Comment:
I dont think we need `_aerIndex ev < 0` any longer as it was guarding against a runtime error in Seq.index where we index under the 0th entry.